lib/readline.js

```diff
@@ -768,6 +780,49 @@ Interface.prototype._moveCursor = function(dx) {
   }
 };
 
+Interface.prototype._ttyWriteDumb = function(s, key) {
+  key = key || {};
+
+  if (key.name === 'escape') return;
+
+  if (this._sawReturnAt && key.name !== 'enter')
+    this._sawReturnAt = 0;
+
+  if (key.ctrl && key.name === 'c') {
+    if (this.listenerCount('SIGINT') > 0) {
+      this.emit('SIGINT');
+    } else {
+      // This readline instance is finished
+      this.close();
+    }
+  }
+
+  switch (key.name) {
+    case 'return':  // carriage return, i.e. \r
+      this._sawReturnAt = Date.now();
+      this._line();
+      break;
+
+    case 'enter':
+      // When key interval > crlfDelay
+      if (this._sawReturnAt === 0 ||
+          Date.now() - this._sawReturnAt > this.crlfDelay) {
+        this._line();
+      }
+      this._sawReturnAt = 0;
+      break;
+
+    default:
+      if (s instanceof Buffer)
+        s = s.toString('utf-8');
```

Comment:
This is currently never triggered and should likely be removed in the original function as well. This function should only be triggered with strings.